### DISCUSSION TITLE
Custom Design or Web UI (File Manager)?

### DISCUSSION THREAD

**Discussion by L2NE-dev** [rejetto/hfs#788] [Ideas]
It's not idea for PR or feature request (will not commited), but how I can something to fully change design of HFS file manager? I have some frontend skills, and thinking about own design and (probably) implementation of UI.

**Comment by rejetto** [comment]
for most part, you should add custom html and CSS in Admin-panel > Custom HTML
https://github.com/rejetto/hfs/wiki/Customization

More is possible using javascript with front-end events
https://github.com/rejetto/hfs/blob/main/dev-plugins.md#front-end-api-events

if you want, you can then package all your changes in a plugin.

**Reply by L2NE-dev** [reply]
What if I would to replace almost whole UI, all icons, all elements, practically... everything?
For example, even write and compile UI in another framework?

**Comment by rejetto** [comment]
if you write another frontend it should be easy for me to offer an option to load its files from some path you provide.
After all, i'm just serving static files from a folder.
If you run the exe version, this folder is embedded, so you can't see it.
If you run the node/npx version, you'll see the "frontend" folder, and can replace it without effort, although this is not a good way to distribute your frontend, just fyi. 

While you develop you probably have your webserver on port 3000 (let's say). In this case you can run the node version setting an env FRONTEND_PROXY=3000 and it will forward the requests to your server. That's how I build my frontend.
